### PR TITLE
Feature/database connection

### PR DESCRIPTION
**Pull request by Caw3**
Updated config to support additional datbase credentials and implements a basic connection to deployment database.

### CODE REVIEW

File: server/src/config.rs [hunk 1]
```diff
@@ -47,9 +48,8 @@ impl Config {
 
         Ok(())
     }
-
-    /// Gets a key value from the trafiklab section in the config file.
-    pub fn get_trafiklab_value(&self, key: &str) -> Option<&str> {
+    /// Gets a key value from the config section in the config file.
+    pub fn get_config_value(&self, field: &str, key: &str) -> Option<&str> {
```
Comment (joelsiks) [inline server/src/config.rs:52]: 👍

File: server/src/database.rs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+use mongodb::{error::Error, options::ClientOptions, Client};
+
+const DATABASE_NAME: &str = "haunterDB";
+
+//Our abstraction for the db, we can use method syntax for operation ex: conn.updateGeoPosition(id, value)
+pub struct Connection {
+    client: Client,
+}
+
+//Inititalise a connection with uri_str
```
Comment (joelsiks) [inline server/src/database.rs]: Same here, change this to a `///` comment.

File: server/src/database.rs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+use mongodb::{error::Error, options::ClientOptions, Client};
```
Comment (joelsiks) [inline server/src/database.rs:2]: Add a `//!` comment as the first line in the file that describes what the file (called module in rust) is used for.

File: server/src/database.rs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+const DATABASE_NAME: &str = "haunterDB";
+
+//Our abstraction for the db, we can use method syntax for operation ex: conn.updateGeoPosition(id, value)
```
Comment (joelsiks) [inline server/src/database.rs]: Give the `Connection` struct a `///` comment so that it has an explanation of what is intended for in the documentation.

**Review by joelsiks — APPROVED** [review]

**Review by caspernorrbin — APPROVED** [review]

**Pull request merged**